### ISSUE TITLE
Error:(3) couldn't find import for class com.fastbootmobile.encore.model.Song

### ISSUE THREAD

**Issue by simioluwatomi** [fastbootmobile/encore#26]
I had the following problems today while trying to build the project. I discovered the model folder in the fastbootmobile.encore folder is missing. Any solution to this?
![image](https://cloud.githubusercontent.com/assets/22191563/23462095/c6865bf2-fe8c-11e6-835d-6514b094a82d.png)

**Comment by xplodwild** [comment]
Make sure you have the providerlib as a module within your project

**Comment by simioluwatomi** [comment]
I guess that was what caused it but the project has a flaw you should fix.

1. When importing the project into Android studio for the first time, it asks if I should allow it use gradle wrapper which would be downloaded from the internet or use my local gradle distribution. Anyone I decide to use throws this error "Error:(1, 0) Plugin with id 'com.android.application' not found.

It shows that the build.gradle file needs to be corrected. Thank you.

![image](https://cloud.githubusercontent.com/assets/22191563/23464094/039b2fc0-fe94-11e6-9c95-f2b87c372983.png)

The error simply needs these lines to be added to the build.gradle file

```
buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.2.3'
    }
}

allprojects {
    repositories {
        jcenter()
        maven { url "https://jitpack.io" }
    }
}
```

After doing that and syncing project, then this error pops up "Error:(78, 0) Project with path ':providerlib' could not be found in root project 'encore'."

![image](https://cloud.githubusercontent.com/assets/22191563/23464340/fcc60b7e-fe94-11e6-960c-2e70b83fd407.png)

Hope you get where I'm coming from.

**Comment by simioluwatomi** [comment]
To solve this, I downloaded the providerlib from here [https://github.com/fastbootmobile/encore-providerlib](url).

Then added it to the  project by creating a settings.gradle file and including the library like so

`include ':library:providerlib'` where `library` is the name of the folder in the root of the project where I copied the providerlib into.

Then in the build.gradle file, I added the dependency like so

`compile project(':library:providerlib')`

I then synced the gradle file and all errors cleared.
